All_allocations

TomCohenAbeille opened the pull request:
feat : compute all allocations 
fix : operating through a DataFrame to open the csv and not working only with lists

TomCohenAbeille committed:
Add files via upload

KwanV92 commented on constraints_2.csv:
On voudrait separer les input / output / code source / code test et les mettre dans des dossiers dédiés

KwanV92 commented on main.py:
```diff
@@ -1,30 +1,45 @@
+    compute_possible_allocations,
+)
+
+os.chdir(
```
A quoi sert cette ligne ?

TomCohenAbeille replied:
rien je l'ai supprimée

KwanV92 commented on constraints.csv:
On voudrait separer les input / output / code source / code test et les mettre dans des dossiers dédiés

KwanV92 commented on allocations.csv:
On voudrait separer les input / output / code source / code test et les mettre dans des dossiers dédiés

KwanV92 commented on dataframe.csv:
On voudrait separer les input / output / code source / code test et les mettre dans des dossiers dédiés

KwanV92 commented on computationals.py:
```diff
@@ -0,0 +1,50 @@
+        detailled_steps.append(asset_possible_allocation)
+    return detailled_steps
+
+def compute_possible_allocations(df:pd.DataFrame) :
```
Que renvoie la fonction ?
Attention a mettre un espace après les deux points lors du typage

KwanV92 commented on computationals.py:
```diff
@@ -0,0 +1,50 @@
+    return detailled_steps
+
+def compute_possible_allocations(df:pd.DataFrame) :
+    ASSETS: tuple = tuple(df.iloc[1].index[1:])
```
les noms des variables sont en minuscules, on réserve les full majuscules pour les constantes

KwanV92 commented on computationals.py:
```diff
@@ -0,0 +1,50 @@
+
+
+def compute_possible_steps(df:pd.DataFrame) ->list : 
+    detailled_steps: list = []
```
naming : detailed_steps avec un seul l

KwanV92 commented on computationals.py:
```diff
@@ -0,0 +1,50 @@
+
+def compute_possible_steps(df:pd.DataFrame) ->list : 
+    detailled_steps: list = []
+    MIN_EXPO: np.ndarray = np.array(df.iloc[0,1:])
+    MAX_EXPO: np.ndarray = np.array(df.iloc[1,1:])
+    STEP: np.ndarray = np.array(df.iloc[2,1:])
```
Que se passe-t-il si les données ne sont pas dans l'ordre min, max, step mais plutot max, step, min ?

KwanV92 commented on computationals.py:
```diff
@@ -0,0 +1,50 @@
+    MAX_EXPO: np.ndarray = np.array(df.iloc[1,1:])
+    STEP: np.ndarray = np.array(df.iloc[2,1:])
+    for index, step in enumerate(STEP):
+        asset_possible_allocation: list = []
```
naming : c'est pas vraiment les allocations possibles, c'est plutot les poids possibles dans l'allocation globale du portfeuille 
Une allocation c'est une somme des poids qui fait 100

KwanV92 commented on computationals.py:
```diff
@@ -0,0 +1,50 @@
+    STEP: np.ndarray = np.array(df.iloc[2,1:])
+    for index, step in enumerate(STEP):
+        asset_possible_allocation: list = []
+        asset_step: int = MIN_EXPO[index]
```
naming : pareil c'est pas vraiment le step de l'asset que tu regardes la, tu es entrain de regarder les différents poids de l'asset

KwanV92 commented on computationals.py:
```diff
@@ -0,0 +1,50 @@
+            asset_possible_allocation.append(asset_step)
+            asset_step = asset_step + step
+        detailled_steps.append(asset_possible_allocation)
+    return detailled_steps
```
naming : pareil ici ce n'est pas vraiment les step que tu renvoies, ce sont plutot les poids possibles de chaque asset

KwanV92 commented on computationals.py:
```diff
@@ -0,0 +1,50 @@
+    allocations_df: pd.DataFrame = pd.DataFrame(columns = ASSETS)
+    detailled_steps: list = compute_possible_steps(df)
+    row_index: int = 0
+    for ASSETS in product(*detailled_steps) : 
```
Attention tu utilises le meme nom pour ton iterateur que pour une variable que tu as defini au dessus, il peut y avoir des confusions ici. Est-ce que tu as voulu réutiliser la variable définie au dessus ?

TomCohenAbeille replied:
oui je réutilise la même variable car je dois avoir un tuple de la même taille que le nombrez d'actifs, d'où l'utilisation de la variable 'assets'

KwanV92 reviewed (changes requested):
Très bien l'algo a l'air de marcher
Quelques problemes de naming qui sont important pour la lecture du code
Une amélioration sur la lecture du dataframe et utilisation de MIN, MAX et STEP listes
Un éclaircissement à me donner sur le passage où tu unpack pour générer les combo

TomCohenAbeille committed:
Add files via upload

feat  : work with the rows in any order 
todo : test if the csv contain all 3 rows (min_expo,max_expo,step)

TomCohenAbeille committed:
Add files via upload

feat : test if the constraints file is valid

TomCohenAbeille committed:
Add files via upload
